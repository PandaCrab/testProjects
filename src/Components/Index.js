import { ShippingInfo } from './Forms/ShippingInfo';
import { BillingInfo } from './Forms/BillingInfo';
import { Payment } from './Forms/Payment';
import { OrderPlate } from './OrderPlate/OrderPlate';


export { ShippingInfo, BillingInfo, Payment, OrderPlate };
