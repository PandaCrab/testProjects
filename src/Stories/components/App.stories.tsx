import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import { ApolloProvider } from '@apollo/client';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

import { store } from '../../redux/store';
import App from '../../App';
import { apolloClient } from '../..';

export default {
    title: 'App',
    component: App
} as ComponentMeta<typeof App>;

const Template: ComponentStory<typeof App> = () => (
    <ApolloProvider client={apolloClient}>
        <Provider store={store}>
            <BrowserRouter>
                <App />
            </BrowserRouter>
        </Provider>
    </ApolloProvider>
);

export const Header = Template.bind({});
