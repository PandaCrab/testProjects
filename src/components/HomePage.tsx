import React from 'react';
import { useNavigate } from 'react-router-dom';

import { 
    StartPage,
    NavigationButton
} from '../styles/HomePageStyles';

const HomePage = () => {
    const navigate = useNavigate();

    return (
        <>
            <StartPage>
                <h1>Hello World</h1>
                <NavigationButton onClick={() => navigate('/shopping')}>Start shopping</NavigationButton>
            </StartPage>
        </>
    );
};

export default HomePage;
