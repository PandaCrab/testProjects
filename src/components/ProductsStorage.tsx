import { useState } from 'react';
import { 
    SortingState,
    IntegratedSorting,
    RowDetailState
} from '@devexpress/dx-react-grid';
import {
  Grid,
  Table,
  TableHeaderRow,
  TableRowDetail,
} from '@devexpress/dx-react-grid-material-ui';
// import { useDispatch, useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { useQuery } from '@apollo/client';

// import { AppDispatch, RootState } from '../types';

// import { getProductsStorage } from '../redux/ducks/stuff';
import { Heading, NavigationButton, StartPage } from '../styles/HomePageStyles';
import { TAKE_PRODUCTS_FROM_STORAGE } from '../api';

interface graphQlData {
    id: string,
    name: string,
    price: number,
    quantity: number
}

const RowDetail = ({ row }: any) => (
    <div>
        {row.quantity > 0 ? 
        `This item can be send today` :
        `Check later, out of stock`
        }
    </div>
)

const ProductsStorage = () => {
    const { error, data } = useQuery(TAKE_PRODUCTS_FROM_STORAGE);
    const navigate = useNavigate();
    // const dispatch = useDispatch<AppDispatch>();
    // const productsStorage = useSelector((state: RootState) => state.order.productsStorage);

    // useEffect(() => { 
    //     dispatch(getProductsStorage())
    // // eslint-disable-next-line react-hooks/exhaustive-deps
    // }, []);

    const [columns] = useState([
        { name: 'id', title: 'Id' },
        { name: 'name', title: 'Name' },
        { name: 'price', title: 'Price' },
        { name: 'quantity', title: 'Quantity' }
    ]);

    const [tableColumnExtensions] = useState([
        { columnName: 'id', width: '10%' },
        { columnName: 'name', width: '50%' },
        { columnName: 'price', width: '10%' },
        { columnName: 'quantity', width: '15%' },
      ]);

    return (
        <StartPage>
            <Heading>
                <h1>Hello in storage</h1>  <NavigationButton onClick={() => navigate('/')} >Home</NavigationButton>
            </Heading>
            <Grid
                rows={data.productsStorage.map(({ id, name, price, quantity }: graphQlData) => ({
                    id,
                    name,
                    price,
                    quantity
                }))}
                columns={columns}
            >
                <SortingState
                    defaultSorting={[{ columnName: 'name', direction: 'asc' }]}
                />
                <IntegratedSorting />
                <RowDetailState />

                <Table columnExtensions={tableColumnExtensions} />
                <TableHeaderRow showSortingControls />
                <TableRowDetail 
                    contentComponent={RowDetail}
                />
            </Grid>
        </StartPage>
    );
};

export default ProductsStorage;
