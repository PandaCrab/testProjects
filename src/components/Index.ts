import ShippingInfo from './forms/ShippingInfo';
import BillingInfo from './forms/BillingInfo';
import Payment from './forms/Payment';
import OrderPlate from './orderPlate/OrderPlate';
import SuccessBuying from './forms/SuccessBuying';
import Navigate from '../Navigate';
import Stuff from './orderPlate/Stuff';
import StuffItems from './orderPlate/StuffItems';
import Loader from './Loader';
import CardToPrint from './forms/CardToPrint';
import DropdownAddresses from './forms/dropdownAddresses';
import CountriesSelect from './forms/CountriesSelect';
import HomePage from './HomePage';
import ProductsStorage from './ProductsStorage';
import SaleChart from './SaleChart';

export { ShippingInfo,
    BillingInfo, 
    Payment, 
    OrderPlate, 
    SuccessBuying, 
    Navigate,
    Stuff,
    StuffItems,
    Loader,
    CardToPrint,
    DropdownAddresses,
    CountriesSelect,
    HomePage,
    ProductsStorage,
    SaleChart
};
