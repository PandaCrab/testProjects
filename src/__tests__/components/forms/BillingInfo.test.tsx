import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import { BillingInfo } from '../../../components/Index';

const initialState: any = {address: {
    navigatorAddress: []
}}
const mockStore = configureStore();
let store;

describe('Billing info component', () => {

    it('should render BillingInfo component correctly', () => {
        store = mockStore(initialState);
        const component = renderer.create(
            <Provider store={store}>
                <BrowserRouter>
                    <BillingInfo />
                </BrowserRouter>
            </Provider>
        );

        expect(component).toMatchSnapshot();
    });

});

describe('Check name field in Billing info component', () => {
    store = mockStore(initialState)
    const component = mount(
        <Provider store={store}>
            <BrowserRouter>
                <BillingInfo />
            </BrowserRouter>
        </Provider>)

    const nameInput = component.find('input[name="name"]')

    it('should have name field', () => {
        expect(nameInput.length).toEqual(1);
    });

    it('should have props', () => {
        expect(nameInput.props()).toEqual({
            className: 'form-control',
            id: 'name',
            name: 'name',
            type: 'text',
            onChange: nameInput.prop('onChange'),
            onBlur: nameInput.prop('onBlur'),
            value: '',
            placeholder: 'Full Name',
            readOnly: undefined,
            size: undefined
        });
    });

    it('should set the name value on change event', () => {
        nameInput.simulate('change', {
            target: { value: 'Vasya Pupkin' } 
        });
        
        const inputAfterChange = component.find('input[name="name"]');
        expect(inputAfterChange.prop('value')).toEqual('Vasya Pupkin');
    });
});

describe('Check email field in Billing info component', () => {
    store = mockStore(initialState)
    const component = mount(
        <Provider store={store}>
            <BrowserRouter>
                <BillingInfo />
            </BrowserRouter>
        </Provider>)

    const emailInput = component.find('input[name="email"]')

    it('should have name field', () => {
        expect(emailInput.length).toEqual(1);
    });

    it('should have props', () => {
        expect(emailInput.props()).toEqual({
            className: 'form-control',
            id: 'email',
            name: 'email',
            type: 'text',
            onChange: emailInput.prop('onChange'),
            onBlur: emailInput.prop('onBlur'),
            value: '',
            placeholder: 'Email address',
            readOnly: undefined,
            size: undefined
        });
    });

    it('should set the name value on change event', () => {
        emailInput.simulate('change', {
            target: { value: 'Kolotushkin@mulo.com' } 
        });
        
        const inputAfterChange = component.find('input[name="email"]');
        expect(inputAfterChange.prop('value')).toEqual('Kolotushkin@mulo.com');
    });
});

describe('check street field', () => {
    store = mockStore(initialState)
    const component = mount(
        <Provider store={store}>
            <BrowserRouter>
                <BillingInfo />
            </BrowserRouter>
        </Provider>)

    const streetInput = component.find('input[name="street"]');

    it('should have street field', () => {
        expect(streetInput.length).toEqual(1);
    });

    it('should have props', () => {
        expect(streetInput.props()).toEqual({
            className: 'form-control',
            id: 'street',
            name: 'street',
            type: 'text',
            onChange: streetInput.prop('onChange'),
            onBlur: streetInput.prop('onBlur'),
            onFocus: streetInput.prop('onFocus'),
            value: '',
            placeholder: 'Street address',
            readOnly: undefined,
            size: undefined
        });
    });

    it('should set the street value on change event', () => {
        streetInput.simulate('change', {
            target: { value: '9 Kolotushina' }
        });

        const inputAfterChange = component.find('input[name="street"]');
        expect(inputAfterChange.prop('value')).toEqual('9 Kolotushina');
    });
});

describe('check optional field', () => {
    store = mockStore(initialState)
    const component = mount(
        <Provider store={store}>
            <BrowserRouter>
                <BillingInfo />
            </BrowserRouter>
        </Provider>)

    const optionalInput = component.find('input[name="optional"]');

    it('should have optional field', () => {
        expect(optionalInput.length).toEqual(1);
    });

    it('should have props', () => {
        expect(optionalInput.props()).toEqual({
            className: 'form-control',
            id: 'optional',
            name: 'optional',
            type: 'text',
            onChange: optionalInput.prop('onChange'),
            onBlur: optionalInput.prop('onBlur'),
            value: '',
            placeholder: 'Apt, Suit, Bidg, Gate Code. (optional)',
            readOnly: undefined,
            size: undefined
        });
    });

    it('should set the optional value on change event', () => {
        optionalInput.simulate('change', {
            target: { value: 'Door code: 3322' }
        });

        const inputAfterChange = component.find('input[name="optional"]');
        expect(inputAfterChange.prop('value')).toEqual('Door code: 3322');
    });
});

describe('check city field', () => {
    store = mockStore(initialState)
    const component = mount(
        <Provider store={store}>
            <BrowserRouter>
                <BillingInfo />
            </BrowserRouter>
        </Provider>)

    const cityInput = component.find('input[name="city"]');

    it('should have city field', () => {
        expect(cityInput.length).toEqual(1);
    });

    it('should have props', () => {
        expect(cityInput.props()).toEqual({
            className: 'form-control',
            id: 'city',
            name: 'city',
            type: 'text',
            onChange: expect.any(Function),
            onBlur: expect.any(Function),
            value: '',
            placeholder: 'City',
            readOnly: undefined,
            size: undefined
        });
    });

    it('should set the city value on change event', () => {
        cityInput.simulate('change', {
            target: { value: 'Laplandia' }
        });

        const inputAfterChange = component.find('input[name="city"]');
        expect(inputAfterChange.prop('value')).toEqual('Laplandia');
    });
});

describe('check zip field', () => {
    store = mockStore(initialState)
    const component = mount(
        <Provider store={store}>
            <BrowserRouter>
                <BillingInfo />
            </BrowserRouter>
        </Provider>)

    const zipInput = component.find('input[name="zip"]');

    it('should have zip field', () => {
        expect(zipInput.length).toEqual(1);
    });

    it('should have props', () => {
        expect(zipInput.props()).toEqual({
            className: 'form-control',
            id: 'zip',
            name: 'zip',
            type: 'text',
            onChange: expect.any(Function),
            onBlur: expect.any(Function),
            value: '',
            placeholder: 'ZIP',
            readOnly: undefined,
            size: undefined
        });
    });

    it('should set the zip value on change event', () => {
        zipInput.simulate('change', {
            target: { value: '321334' }
        });

        const inputAfterChange = component.find('input[name="zip"]');
        expect(inputAfterChange.prop('value')).toEqual('321334');
    });

    describe('Submit button', () => {
        store = mockStore(initialState);
        const component = mount(
            <Provider store={store}>
                <BrowserRouter>
                    <BillingInfo />
                </BrowserRouter>
            </Provider>
        );
    
        const button = component.find('button[type="submit"]');
    
        it('should have submit button', () => {
            expect(button.length).toEqual(1);
        });
    
        it('should submit on click', () => {
            const mockHandleSubmit = jest.fn();
            component.find('form').prop('onSubmit')(mockHandleSubmit());
            button.simulate('click');
    
            expect(mockHandleSubmit).toHaveBeenCalled();
        });
    });
});
