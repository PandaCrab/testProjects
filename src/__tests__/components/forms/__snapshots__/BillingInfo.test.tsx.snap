// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Billing info component should render BillingInfo component correctly 1`] = `
Array [
  <section
    className="sc-cOFTSb foYTCl"
  >
    <a
      href="/shipping"
      onClick={[Function]}
    >
      Shipping
    </a>
     &gt;
    <a
      href="/billing"
      onClick={[Function]}
    >
      Billing
    </a>
     &gt;
    <a
      href="/payment"
      onClick={[Function]}
    >
      Payment
    </a>
  </section>,
  <div
    className="sc-bczRLJ iLjvJl"
  >
    <section
      className="sc-jqUVSM drOJkA"
    >
      <h4
        className="sc-kDDrLX hxhBIM"
      >
        Billing Information
      </h4>
      <p
        className="sc-iqcoie irDhwn"
        onClick={[Function]}
      >
        Same as shipping
      </p>
    </section>
    <form
      className=""
      noValidate={true}
      onSubmit={[Function]}
    >
      <div>
        <label
          className="mb-0 form-label"
        >
          Billing Contact
        </label>
        <div
          className="mb-4 position-relative"
        >
          <input
            className="form-control"
            id="name"
            name="name"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Full Name"
            type="text"
            value=""
          />
          <div
            className="invalid-tooltip"
          />
        </div>
        <div
          className="mb-4 mb-md-4 mb-lg-5 mb-xl-5 position-relative"
        >
          <input
            className="form-control"
            id="email"
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Email address"
            type="text"
            value=""
          />
          <div
            className="invalid-tooltip"
          />
        </div>
      </div>
      <div>
        <div
          className="position-relative"
        >
          <label
            className="mb-0 form-label"
          >
            Billing address
          </label>
          <div
            className="mb-4 position-relative"
          >
            <input
              className="form-control"
              id="street"
              name="street"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Street address"
              type="text"
              value=""
            />
            <div
              className="invalid-tooltip"
            />
          </div>
          <div
            className="mb-2 position-relative"
          >
            <input
              className="form-control"
              id="optional"
              name="optional"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Apt, Suit, Bidg, Gate Code. (optional)"
              type="text"
              value=""
            />
          </div>
          <div
            className="mb-4 position-relative"
          >
            <input
              className="form-control"
              id="city"
              name="city"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="City"
              type="text"
              value=""
            />
            <div
              className="invalid-tooltip"
            />
          </div>
          <div
            className="mb-4 position-relative"
          >
            <div
              className="row"
            >
              <div
                className="col-sm-7 col-7"
              >
                <div
                  className=" css-b62m3t-container"
                  id="country"
                  onKeyDown={[Function]}
                >
                  <span
                    className="css-1f43avz-a11yText-A11yText"
                    id="react-select-9-live-region"
                  />
                  <span
                    aria-atomic="false"
                    aria-live="polite"
                    aria-relevant="additions text"
                    className="css-1f43avz-a11yText-A11yText"
                  />
                  <div
                    className=" css-1s2u09g-control"
                    onMouseDown={[Function]}
                    onTouchEnd={[Function]}
                  >
                    <div
                      className=" css-319lph-ValueContainer"
                    >
                      <div
                        className=" css-14el2xx-placeholder"
                        id="react-select-9-placeholder"
                      >
                        Country
                      </div>
                      <div
                        className=" css-6j8wv5-Input"
                        data-value=""
                      >
                        <input
                          aria-autocomplete="list"
                          aria-describedby="react-select-9-placeholder"
                          aria-expanded={false}
                          aria-haspopup={true}
                          autoCapitalize="none"
                          autoComplete="off"
                          autoCorrect="off"
                          className=""
                          disabled={false}
                          id="react-select-9-input"
                          onBlur={[Function]}
                          onChange={[Function]}
                          onFocus={[Function]}
                          role="combobox"
                          spellCheck="false"
                          style={
                            Object {
                              "background": 0,
                              "border": 0,
                              "color": "inherit",
                              "font": "inherit",
                              "gridArea": "1 / 2",
                              "label": "input",
                              "margin": 0,
                              "minWidth": "2px",
                              "opacity": 1,
                              "outline": 0,
                              "padding": 0,
                              "width": "100%",
                            }
                          }
                          tabIndex={0}
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                    <div
                      className=" css-1hb7zxy-IndicatorsContainer"
                    >
                      <span
                        className=" css-1okebmr-indicatorSeparator"
                      />
                      <div
                        aria-hidden="true"
                        className=" css-tlfecz-indicatorContainer"
                        onMouseDown={[Function]}
                        onTouchEnd={[Function]}
                      >
                        <svg
                          aria-hidden="true"
                          className="css-tj5bde-Svg"
                          focusable="false"
                          height={20}
                          viewBox="0 0 20 20"
                          width={20}
                        >
                          <path
                            d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                  <input
                    name="country"
                    type="hidden"
                    value=""
                  />
                </div>
              </div>
              <div
                className="col-sm-5 col-5"
              >
                <input
                  className="form-control"
                  id="zip"
                  name="zip"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="ZIP"
                  type="text"
                  value=""
                />
                <div
                  className="invalid-tooltip"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <button
        className="sc-papXJ fRWQUH btn btn-secondary"
        disabled={false}
        type="submit"
      >
        Continue
      </button>
    </form>
  </div>,
]
`;
